File: joiner-core/src/test/java/cz/encircled/joiner/model/TestEntity.java
package cz.encircled.joiner.model;

import javax.persistence.Column;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

/**
 * @author Kisel on 21.01.2016.
 */
@MappedSuperclass
public class TestEntity {

    @Id
    @Column(name = "id")
    protected Long id;

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

}

File: joiner-ksp/src/test/kotlin/cz/encircled/joiner/ksp/TestEntity.kt
package cz.encircled.joiner.ksp

import com.querydsl.core.annotations.QueryTransient
import jakarta.persistence.*

/**
 * A simple test entity for testing the KSP processor.
 */
@Entity
@Table(name = "test_entity")
class TestEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "name")
    var name: String? = null

    @Column(name = "active")
    var active: Boolean = false

    @OneToMany(mappedBy = "parent")
    var children: Set<TestEntity>? = null

    @ManyToOne
    @JoinColumn(name = "parent_id")
    var parent: TestEntity? = null

    @ManyToOne
    val testEntity: TestEntity? = null // self ref with the same name

    @Transient
    var transientField: String? = null

    @QueryTransient
    var transientField2: String? = null

    @get:Transient
    val transientFieldWithCustomGetter get() = "Custom Getter"

}